src/uct/cuda/base/cuda_iface.c

```diff
@@ -81,6 +85,17 @@ ucs_status_t uct_cuda_base_iface_event_fd_get(uct_iface_h tl_iface, int *fd_p)
     return UCS_OK;
 }
 
+ucs_status_t uct_cuda_base_check_device_name(const uct_iface_params_t *params)
+{
+    if (strncmp(params->mode.device.dev_name, UCT_CUDA_DEV_NAME,
```

Comment:
i think we also need to check `params->field_mask & UCT_IFACE_PARAM_FIELD_DEVICE`